app/src/main/java/org/mozilla/reference/browser/concepts/HistoryStorage.kt

```diff
@@ -29,13 +29,46 @@ interface HistoryStorage : HistoryStorage {
      */
     fun restoreTopSites()
 
+    /**
+     * Gets all the bookmarks.
+     * @return a list of [BookmarkNode]
+     */
```

Comment:
In order to make things easier also for Ghostery, why don't you just extend this interface with the `BookmarksStorage` concept instead of reinventing the wheel?

Reply:
I made the changes so that `HistoryDatabase` directly implements the `BookmarksStorage` of components cause there's nothing new our `BookmarksStorage` interface was doing.

And now, `GetBookmarksUseCase` fetches a node of a tree with bookmarks in it rather than a list of bookmarks now.